### PR TITLE
Make presenters work with the api

### PR DESCRIPTION
**Pull request by joecohens**
This is just a proof of concept of how can we make the presenters work with the API responses. What do you think @jbrooksuk @GrahamCampbell ?

Also this fixes two bugs on this PR https://github.com/cachethq/Cachet/issues/625

### CODE REVIEW

File: app/Http/Controllers/Api/AbstractApiController.php [hunk 1]
```diff
@@ -35,9 +38,9 @@
     /**
      * The HTTP response data.
      *
-     * @var array
+     * @var mixed
```
Comment (GrahamCampbell) [inline app/Http/Controllers/Api/AbstractApiController.php]: mixed should be avoided when possible

**Comment by jbrooksuk** [comment]
Looks great to me!

File: app/Http/Controllers/Api/AbstractApiController.php (hunk 1 above)
Comment (joecohens) [inline app/Http/Controllers/Api/AbstractApiController.php]: Yeah you are right, do you think of a better way to accept the content data?

File: app/Config/Repository.php [hunk 2]
```diff
@@ -45,11 +45,12 @@ public function __construct(Setting $model)
      * Returns a setting from the database.
      *
      * @param string $name
+     * @param string $default
```
Comment (GrahamCampbell) [inline app/Config/Repository.php]: string|null

File: app/Presenters/MetricPointPresenter.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+use Jenssegers\Date\Date;
+
+class MetricPointPresenter extends AbstractPresenter
+{
+    use TimestampsTrait;
+
+    /**
+     * Convert the presenter instance to an array.
+     *
+     * @return array
```
Comment (GrahamCampbell) [inline app/Presenters/MetricPointPresenter.php]: can we be more specific here?

File: app/Presenters/MetricPresenter.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+use Jenssegers\Date\Date;
+
+class MetricPresenter extends AbstractPresenter
+{
+    use TimestampsTrait;
+
+    /**
+     * Convert the presenter instance to an array.
+     *
+     * @return array
```
Comment (GrahamCampbell) [inline app/Presenters/MetricPresenter.php]: and in all places :)

File: app/Presenters/MetricPointPresenter.php (hunk 3 above)
Comment (joecohens) [inline app/Presenters/MetricPointPresenter.php]: Can you give me an example?

Comment (GrahamCampbell) [inline app/Presenters/MetricPointPresenter.php]: `string[]`?

Comment (joecohens) [inline app/Presenters/MetricPointPresenter.php]: Lol ok :)

**Comment by joecohens** [comment]
@jbrooksuk All done :) merge at will.

**Comment by jbrooksuk** [comment]
Yes! :D

**Comment by jbrooksuk** [comment]
:dancer:

**Pull request merged**